### ISSUE TITLE
Wysiwyg Visual Tab is Blank, and the mouse cannot be positioned to the Editor, and the content cannot be edited

### ISSUE THREAD

**Issue by sunbbon88** [CMB2/CMB2#1397]
### Describe the bug
When I add custom metabox using a wysiwyg type field, Only add data  at the Text Tab. When I switch to the Visual Tab, the content is Blank, and the mouse cannot be positioned to the Visual Editor, and the content cannot be edited.

### CMB2 Field Registration Code:

`$cmb_product->add_field( array(
		'name'    => esc_html__( 'Product Description', 'wppopMetabox' ),
		'desc'    => esc_html__( '', 'wppopMetabox' ),
		'id'      => 'product_excerpt',
		'type'    => 'wysiwyg',
		'sanitization_cb' => false,
		'options' => array(
			'textarea_rows' => 25,
		),
	) );`

### Screenshots

This is Text Tab:
https://i.postimg.cc/ZKjBGYHJ/20201231085623.png

SameTime, switch to Visual Tab:
https://i.postimg.cc/tC9nvnhr/20201231085612.png

**Comment by tw2113** [comment]
Can you provide some more information.

Specifically which version of WordPress you're presently on, as well as which version of CMB2 you're using?

I'm not managing to recreate this at the moment. I can edit text in either tab, using WordPress 5.6 and CMB2 2.7.0

**Comment by sunbbon88** [comment]
WordPress: 5.6
CMB2: 2.7.0
PHP: 7.4.12

This problem occurred when using the Gutenberg editor. When I used the Classic Editor, it worked normally.

**Comment by tw2113** [comment]
hmm only difference from what I tested with a little bit ago is that the PHP version is different, and that likely doesn't have any real effect on this.

Do you have a test environment you could use where you only have CMB2 active for a plugin? I'm curious if you have a different conflict going on, or maybe some javascript errors occurring that I'm not seeing.

**Comment by sunbbon88** [comment]
Thank you for your reply.
I rebuilt an original WordPress site and tested CMB2 with twentytwentyone theme, This problem still exists. Other types of fields are normal. Only when using Gutenberg editor, wysiwyg type field with Visual Tab has a problem.
Moreover, under Chrome browser (Version: 87.0.4280.88) and Firefox browser (Version: 84.0.1), there is no javascript error prompt.

=====================

I personally think that browser and PHP version problems can be ruled out, because I have encountered the same problem in different browsers and PHP versions.

**Comment by tw2113** [comment]
Even with TwentyTwentyOne I'm not managing to create this.

I do know there's a number of jQuery/Javascript issues going on with WordPress 5.6.0, I'm wondering if maybe those are contributing in some way, and once 5.6.1 is out the issue will be gone.

**Comment by sunbbon88** [comment]
I have tested it with WordPress 5.6 and WordPress 5.5.3 in exactly the same environment. This problem only appears in WordPress 5.6, and there is no problem in WordPress 5.5.3.
I hope that this problem can be solved after WordPress is upgraded to 5.6.1.

**Comment by tw2113** [comment]
Here's to hoping

**Comment by lumpysimon** [comment]
I'm seeing this same issue in WordPress 5.6 while using the block editor: the WYSIWYG editor is unusable in visual mode and all content in it is invisible. If I switch to text mode, the content is there. If I use the classic editor, everything works fine.

**Comment by pgrab86** [comment]
I'm having the same issue with Wordpress 5.6. I tried to install Gutenberg as a plugin but it didn't work as well. Found a solution - switch to text mode, refresh page - you should see editor in text mode. Now switch back to WYSIWYG mode - it loads fine. But it doesn't work on fields in repeatable groups. Looks, to me, like there is no initial action for the TinyMCE.

**Comment by tw2113** [comment]
Out of curiosity, has anyone tried things with a trunk copy of WP Core to see if the issues experienced have gone away? Would be awesome to have some verification that 5.6.1 will resolve the experienced issues

**Comment by jeremycaris** [comment]
I just found this thread because I'm getting the same issue coding from scratch. Here's my post on [Stackexchange](https://wordpress.stackexchange.com/questions/382306/wp-editor-visual-tab-in-meta-box-doesnt-show-content/382310#382310). I found CMB2 and implemented it hoping to resolve the issue, but got the same result.

**Comment by tw2113** [comment]
Hi @jeremycaris since you were experiencing the same with your custom rolled solution, that's re-enforcing my hunches that this is coming from WP core and not necessarily CMB2.

Decided to do a quick Core Trac search and found https://core.trac.wordpress.org/ticket/52050

However, as of 73min ago from the time of my post here, it just got bumped to WordPress 5.6.2 instead of 5.6.1, so we will have more waiting time for possible solutions here.

We definitely use wp_editor() in our WYSIWYG field type.

**Comment by jeremycaris** [comment]
Update... I found that if you install the Classic Editor plugin to turn off Gutenberg, it works just fine. So it's definitely an issue with Gutenberg.

**Comment by sunbbon88** [comment]
> Update... I if you install the Classic Editor plugin to turn off Gutenberg, it works just fine. Or, if you need to leave Gutenberg active, installing the Enable jQuery Migrate Helper plugin resolves the issue by downgrading jQuery back to the version previous included in Wordpress 5.5.

Thank you very much for your reply and **your solution**!

The **Enable jQuery Migrate Helper** plugin can indeed solve this problem  when using Gutenberg editor.

But, Unfortunately, this problem still exists in WordPress 5.6.1. I have tested other similar Custom Metabox plug-ins, and it seems that there is no such problem. I guess it may be related to CMB2's compatible new version of jQuery.

**Comment by sagarsdeshmukh** [comment]
A temporary workaround which works for me is, make a 'text' mode default for wp-editor, like

```
// Set wp-editor in the 'Text' mode by default.
add_filter(
	'wp_default_editor',
	function () {
		return 'html';
	}
);
```
When we switch from text to visual, the wp-editor(tinyMCE) works fine, though its not a perfect fix but a quick win over adding a whole new plugin(Enable jQuery Migrate Helper).

**Comment by sunbbon88** [comment]
> A temporary workaround which works for me is, make a 'text' mode default for wp-editor, like
> 
> ```
> // Set wp-editor in the 'Text' mode by default.
> add_filter(
> 	'wp_default_editor',
> 	function () {
> 		return 'html';
> 	}
> );
> ```
> 
> When we switch from text to visual, the wp-editor(tinyMCE) works fine, though its not a perfect fix but a quick win over adding a whole new plugin(Enable jQuery Migrate Helper).

Thanks to sagarsdeshmukh for the solution, this is the relatively most perfect way to deal with it so far.

**Comment by sagarsdeshmukh** [comment]
Good news about this issue, soon the fix will be rollout by WP 🎉 
The [issue](https://core.trac.wordpress.org/ticket/52050) is closed, and on the way to release with [WPv5.8](https://core.trac.wordpress.org/query?status=closed&group=resolution&milestone=5.8) 

As per the [roadmap](https://wordpress.org/about/roadmap/), the WPv5.8 will be released by July 2021.

**Comment by sunbbon88** [comment]
This issue has been solved in WordPress 5.8.
